tasks/codeblock-jshint.js

```diff
@@ -2,93 +2,68 @@
 
 var path = require( 'path' );
 var JSHint = require( 'jshint' ).JSHINT;
-var defaultReporter = require( 'jshint/src/reporters/default' ).reporter;
-var marked = require( 'marked' );
+var tokenizeMarkdown = require( 'tokenize-markdown' );
+var contribJSHint = require( './lib/jshint' );
 
 module.exports = function( grunt ) {
   var desc = 'Run JSHint against code snippets within Markdown slides.';
 
-  grunt.registerTask( 'codeblock-jshint', desc, function codeBlockJSHintTask() {
+  grunt.registerMultiTask( 'codeblock-jshint', desc, function codeBlockJSHintTask() {
+    var done = this.async();
+
     // Get any configured options, defaulting the search paths to any markdown
     // files below the working directory other than those in node_modules
     var options = this.options({
-      src: [
-        '**/*.md',
-        '!node_modules/**/*.md'
-      ]
+      force: false
     });
 
-    var reporter;
-
-    // Attempt to apply a provided reporter
-    if ( ! options.reporter ) {
-      // If no configuration was supplied, use the default
-      reporter = defaultReporter;
-    } else if ( typeof options.reporter === 'string' ) {
-      // Otherwise, if a string was provided, attempt to load the reporter from
-      // a module path string in the manner that grunt-contrib-jshint does
-      options.reporter = path.resolve( process.cwd(), options.reporter );
-      try {
-        reporter = require( options.reporter ).reporter;
-      } catch ( err ) {
-        grunt.fatal( err );
-      }
-    } else if ( typeof options.reporter === 'function' ) {
-      // Handle cases like `reporter: function myCustomReporter() {}`
-      reporter = options.reporter;
-    } else {
-      // Fall back to the default reporter
-      reporter = defaultReporter;
-    }
-
-    // Process all slides markdown files and turn into token lists from marked.
-    function convertToTokens( file ) {
-      var contents = grunt.file.read( file );
-      var tokens = marked.lexer( contents );
+    // If force is true, report JSHint errors but dont fail the task
+    var force = options.force;
+    delete options.force;
 
-      return {
-        file: file,
-        tokens: tokens
-      };
-    }
+    // Use logic borrowed from grunt-contrib-jshint to select the reporter to use
+    var reporter = contribJSHint.init( grunt ).selectReporter( options );
 
-    var markdownTokens = grunt.file.expand( options.src ).map( convertToTokens );
-
-    // Utility for filtering out tokens that don't represent code blocks
-    function onlyJSTokens( token ) {
-      return token.type === 'code' && token.lang === 'javascript';
-    }
+    // Process all slides markdown files and turn into token lists from marked,
+    // filtering the returned tokens down to only JS files
+    var markdownTokens = tokenizeMarkdown.fromFiles( this.filesSrc, {
+      type: 'code',
+      lang: /(js|javascript)/
```

Comment:
Do you think this could be a good place to branch and allow skips? If you wanted to just display the syntax without linting, use ``` js?